.github/workflows/main.yml

```diff
@@ -29,19 +40,19 @@ jobs:
         with:
           dotnet-version: 6.0.x
 
+      # Adjustment is done prior to Nuke build as OCTOVERSION information is included in the result package.
+      - name: Append OCTOVERSION_CurrentBranch with -nightly (for scheduled)
+        if: github.event_name == 'schedule'
+        run: echo "OCTOVERSION_CurrentBranch=${{ env.OCTOVERSION_CurrentBranch }}-nightly" >> $env:GITHUB_ENV
+
       - name: Nuke Build 🏗
         id: build
         shell: bash
-        # Pass branch and patch number to Nuke OctoVersion
-        # (for pull_request events we override the /refs/pull/xx/merge branch to the PR's head branch)
-        env:
-          OCTOVERSION_CurrentBranch: ${{ github.head_ref || github.ref }}
-          OCTOVERSION_Patch: ${{ github.run_number }}
         run: ./build.sh
         
       - name: Tag release (when not pre-release) 🏷️
         id: github-tag
-        if: ${{ !contains( steps.build.outputs.octoversion_fullsemver, '-' ) }}
+        if: ${{ github.event_name != 'schedule' && !contains( steps.build.outputs.octoversion_fullsemver, '-' ) }}
```

Comment:
If we're appending a prerelease tag above, do we need this extra check on the trigger?

Reply:
Yes, we don't want to create a tag for nightly builds.

Reply:
Yep, absolutely. We're adding `-nightly` as a prerelease tag, which the `!contains( steps.build.outputs.octoversion_fullsemver, '-' )` condition would catch, wouldn't it?

Reply:
Ah, I've misread this step... I thought this was creating tags for pre-release builds too 🤦
I'm happy to remove it, though I like that it's consistent with the other actions we have, e.g: [OctopusClients](https://github.com/OctopusDeploy/OctopusClients/blob/1e047d18c7ae725b6d9583153137fa04252fb914/.github/workflows/main.yaml#L92) and [Octodiff](https://github.com/OctopusDeploy/Octodiff/blob/e3bc41e28d33c0c921b3ada0a06b60a64561662b/.github/workflows/main.yml#L84).

No strong opinions either way.

Reply:
Keeping it consistent sounds good. It's more explicit this way, too.